java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/project/ProjectAlertPresenter.java

```diff
@@ -0,0 +1,622 @@
+    /**
+     * Overall result. The future will complete after all errors were seen, all questions were answered or timed out.
+     */
+    final CompletableFuture<Boolean> completion;
+    
+    /**
+     * All errors already acknowledged by the process, but not necessarily presented
+     */
+    // @GuardedBy(this)
+    Set<BrokenReferencesModel.ProblemReference> snapshot = Collections.emptySet();
+    
+    /**
+     * Invocation attempt, i.e. after a timeout or new project alert. The process continues after user response only
+     * if another run attempt has not started yet.
+     */
+    // @GuardedBy(this)
+    int runAttempt;
+
+    /**
+     * If true, will re-display project problems that have been already presented to the client.
+     */
+    boolean includeSeen;
+
+    /**
+     * Timer task that will push the process further if the user ignores the prompts. The task is run by {@link #RP} processor.
+     */
+    // @GuardedBy(this)
+    private RequestProcessor.Task wakeUpTask;
+    
+    /**
+     * The resolution process is cancelled. The process will terminate after the first timeout or user input.
+     */
+    // @GuardedBy(this)
+    private boolean cancelled;
+    
+    /**
+     * Indicates that all questions were answered. Will eventually reset to {@code false} during the process, if the user
+     * cancels further fixes.
+     */
+    // @GuardedBy(this)
+    boolean allProcessed = true;
+    
+    /**
+     * Wait threads that only record user's acknowledge for a fatal error. No real interaction is required, so there can be more
+     * of these messages to confirm
+     */
+    // @GuardedBy(RP.this)
+    RequestProcessor fatalMsgs = new RequestProcessor(ProjectAlertPresenter.class.getName(), MAX_PRESENTED_ERRORS);
+    
+    /**
+     * Question threads that wait on user input. Each of the threads wait for the user to ask the yes/no/rest question.
+     */
+    // @GuardedBy(RP.this)
+    RequestProcessor questions = new RequestProcessor(ProjectAlertPresenter.class.getName(), MAX_PRESENTED_QUESTIONS);
+    
+    public ProjectAlertPresenter(Project project, BrokenReferencesModel model, final Env master) {
+        this.controller = master;
+        this.project = project;
+        this.model = model;
+        ProjectInformation pi = ProjectUtils.getInformation(project);
+        String pn = pi.getDisplayName();
+        if (pn == null) {
+            pn = pi.getName();
+        }
+        this.projectName = pn;
+        LOG.log(Level.FINE, "Initializing alert presenter for {0}", projectName);
+        completion = new CompletableFuture<Boolean>() {
+            @Override
+            public boolean cancel(boolean mayInterruptIfRunning) {
+                boolean r = super.cancel(mayInterruptIfRunning);
+                if (r) {
+                    ProjectAlertPresenter.this.cancel();
+                }
+                return r;
+            }
+        };
+        model.addListDataListener(new ListDataListener() {
```

Comment:
Using Swing in headless module is probably not appropriate.

Reply:
Well... it's weird, but the data model does not use any graphics-related stuff.